particles/Tutorial/Javascript/Demo/source/MoveApplier.js

```diff
@@ -0,0 +1,104 @@
+/**
+ * @license
+ * Copyright 2019 Google LLC.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * Code distributed by Google as part of this project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+ /* global defineParticle */
+
+ defineParticle(({DomParticle, html}) => {
+
+  return class extends DomParticle {
+    get template() {
+      return '';
+    }
+
+    update({move, gameState}, {}) {
+      if (!gameState) { // Gamestate has not been set yet
+        const gs = {
+          board: ',,,,,,,,',
+          moves: 0,
+          gameOver: false,
+          winnerAvatar: null,
+          attemptedMoves: 0
+        };
+        this.updateSingleton('gameState', gs);
+      } else if (move) {
+        // Get the old gameState values to update.
+        let newMoves = gameState.moves;
+        let newBoard = gameState.board;
+        let newAttemptedMoves = gameState.attemptedMoves;
+        let newGameOver = gameState.gameOver;
+        let newWinnerAvatar = gameState.winnerAvatar;
+
+        // Get the board and move in a usable state
+        const arr = gameState.board.split(`,`);
+        const mv = parseInt(move.move, 10) - 1;
```

Comment:
We could. I initially made it a string so you could more easily extend this for other games. The idea is the board _could_ be replaced with another game board (for example chess) and then moves are probably better expressed as strings. (for example in chess you could have "Be5"). This probably should be changed in line with my rant about designing the tutorial for what we want it to do, not what it could do.

Reply:
There are now "start" and "reset" as moves, so a string is handy :)